sqlite_query.c

```diff
@@ -43,10 +45,7 @@ sqlite_convert_to_pg(Oid pgtyp, int pgtypmod, sqlite3_stmt * stmt, int stmt_coli
 
 	if (affinity_for_pg_column != sqlite_value_affinity && sqlite_value_affinity == SQLITE3_TEXT)
 	{
-		/* TODO: human readable error message based on this code
-		 * elog(ERROR, "SQLite data affinity = \"%s\" disallowed for PostgreSQL type \"%s\" = SQLite \"%s\", value ='%s'", sqlite_affinity, pg_dataTypeName, pg_eqv_affinity, (char*)value_datum);
-		 */
-		elog(ERROR, "invalid input syntax for type =%d, column type =%d", affinity_for_pg_column, sqlite_value_affinity);
+		sqlite_value_to_pg_error (pgtyp, pgtypmod, stmt, attnum, sqlite_value_affinity, affinity_for_pg_column, value_byte_size_blob_or_utf8);
```

Comment:
The error message was change, could you update the test result?

Reply:
I have tried in https://github.com/mkgrgis/sqlite_fdw/commit/0beab63bf98773b76e37b5bb3b0d59aea945e7c5 , @nxhai98. Now I am near reproducing multi-versional testing system of `sqlite_fdw`, but yet I am unable fully observe all tests. Look like my new values of outputs are correct.

Reply:
I run on my env and the test failed:
```diff
+++ /home/workplace/postgresql-15.0/contrib/sqlite_fdw/results/15.0/extra/sqlite_fdw_post.out	2023-05-25 10:04:20.318002000 +0700
@@ -4954,16 +4954,16 @@
 ALTER FOREIGN TABLE ft1 ALTER COLUMN c8 TYPE int;
 --Testcase 273:
 SELECT * FROM ft1 ftx(x1,x2,x3,x4,x5,x6,x7,x8) WHERE x1 = 1;
-ERROR:  SQLite data affinity "text" disallowed for PostgreSQL data type "int" = SQLite "int", value = 'foo'
+ERROR:  SQLite data affinity "text" disallowed for PostgreSQL data type "integer" = SQLite "integer", value = '(null)'
```
About `value = '(null)'`, I think it because `attnum` arg it used to get data value from sqlite statement in sqlite_value_to_pg_error ,
=> it should be `stmt_colid`.

Could you check for this?

Reply:
> Could you check for this?

Yes. I am sorry for `attnum`, because it was your interesting commit https://github.com/pgspider/sqlite_fdw/commit/dbb7a390c2c6b2aeaa543777391f1dbc7e2d0401 with this fix, but I forget new style for `sqlite3_column_`something.
May be fixed in https://github.com/pgspider/sqlite_fdw/pull/75/commits/2fa07fed648a184f03e25a0146683c2a09b60f6b. Also `int` changed to `integer`.

Reply:
Thank for fixing. I confirmed it.

Reply:
Has this review successfully ended, @nxhai98 ?

Reply:
The review is done. I verify again by run all versions test please check https://github.com/pgspider/sqlite_fdw/pull/75#issuecomment-1562505754